boa_engine/Cargo.toml

```diff
@@ -57,7 +57,7 @@ fast-float = "0.2.0"
 unicode-normalization = "0.1.19"
 dyn-clone = "1.0.4"
 once_cell = "1.9.0"
-
+tap = "1.0.1"
```

Comment:
I don't think we really need this dependency, Since all we use it is to convert a value to a `Option`

Reply:
I think it adds clarity to the pipeline associated with the builtin constructors.
```rust
ConstructorBuilder::new()
...
... several lines of operations
...
    .build()
    .conv::<JsValue>()
    .pipe(Ok)
```
is, for me, a lot clearer than

```rust
let object = ConstructorBuilder::new()
...
... several lines of operations
...
        .build();
Ok(object.into())
```

Also, the dependency is pretty lightweight, is widely used in the Rust community (7M downloads) and it doesn't use any other dependency, which usually is our main reason against adding a dependency.